services/cd-service/pkg/service/lock_test.go

```diff
@@ -229,24 +238,90 @@ func TestInvalidCreateEnvironmentApplicationLockArguments(t *testing.T) {
 		tc := tc
 		t.Run(tc.Name, func(t *testing.T) {
 			t.Parallel()
-			svc := &LockServiceServer{}
-			_, err := svc.CreateEnvironmentApplicationLock(
-				context.Background(),
-				tc.Request,
-			)
-			if err == nil {
-				t.Fatal("expected an error, but got none")
-			}
-			if s, ok := status.FromError(err); !ok {
-				t.Fatalf("expected a status error, but got %#v", err)
-			} else {
-				if s.Code() != tc.ExpectedCode {
-					t.Errorf("invalid error code: expected %q, actual: %q", tc.ExpectedCode.String(), s.Code().String())
+			logs := testlogger.Wrap(context.Background(), func(ctx context.Context) {
+				svc := &LockServiceServer{}
+				_, err := svc.CreateEnvironmentApplicationLock(
+					ctx,
+					tc.Request,
+				)
+				if err == nil {
+					t.Fatal("expected an error, but got none")
 				}
-				if s.Message() != tc.ExpectedMessage {
-					t.Errorf("invalid error message: expected %q, actual: %q", tc.ExpectedMessage, s.Message())
+				if s, ok := status.FromError(err); !ok {
+					t.Fatalf("expected a status error, but got %#v", err)
+				} else {
+					if s.Code() != tc.ExpectedCode {
+						t.Errorf("invalid error code: expected %q, actual: %q", tc.ExpectedCode.String(), s.Code().String())
+					}
+					if s.Message() != tc.ExpectedMessage {
+						t.Errorf("invalid error message: expected %q, actual: %q", tc.ExpectedMessage, s.Message())
+					}
 				}
-			}
+			})
+			testlogger.AssertEmpty(t, logs)
+		})
+	}
+}
+
+func TestErrorLock(t *testing.T) {
+	tcs := []struct {
+		Name            string
+		Request         *api.CreateEnvironmentApplicationLockRequest
+		ExpectedCode    codes.Code
+		ExpectedMessage string
+	}{
+		{
+			Name: "create an environment lock",
+			Request: &api.CreateEnvironmentApplicationLockRequest{
+				Environment: "bar",
+				Application: "app",
+				LockId:      "lock",
+			},
+			ExpectedCode:    codes.Internal,
+			ExpectedMessage: "internal error",
```

Comment:
hm, could we provide a more meaningful error message here?
It's not clear to me, what is actually tested (or: why do we get this error here)

Reply:
:thinking: reading this now, I think that shouldn't be tested in the test case at all.